plugins/generator-1.19.4/forge-1.19.4/procedures/entity_loot_table_foreach.java.ftl

```diff
@@ -0,0 +1,23 @@
+<#include "mcelements.ftl">
+<#-- @formatter:off -->
+if (${input$entity} instanceof LivingEntity _entLootTbl && !_entLootTbl.level.isClientSide() && _entLootTbl.getServer() != null) {
+	DamageSource _dsLootTbl = _entLootTbl.getLastDamageSource();
+	if (_dsLootTbl == null) _dsLootTbl = new DamageSource(_entLootTbl.level.registryAccess().registryOrThrow(Registries.DAMAGE_TYPE).getHolderOrThrow(DamageTypes.GENERIC));

```

Comment:
A much nicer approach for this would be eg. `_entLootTbl.damageSources().generic()`

Reply:
Why didn't you use it for the `deal_damage` block?

Reply:
Because mappings work by accessing "DamageTypes" type, not defining method calls eg. generic().

The fallback in this method is only called in rare cases of old workspaces where damage type field did not exist and thus it is arguably cleaner to just provide equal fallback in DamageTypes format

Reply:
Is it fine if I use `new DamageSource(new DamageType("generic", 0.0f))`?

Reply:
Why do you refuse to use the method provided by Mojang?

Reply:
Besides, your approach is making a new DamageType instance that should be registered in Registries.DAMAGE_TYPE

Reply:
Don't get me wrong, but I really fail to see how this conversation is going in a productive way and makes sense (or justifies time spent on it)

Reply:
>Why do you refuse to use the method provided by Mojang?

Erm... I'm not aware where I could overlook it?

Reply:
> A much nicer approach for this would be eg. _entLootTbl.damageSources().generic()

Reply:
I provided a copy-paste solution in my comment afaik